static/app/views/settings/organizationTeams/teamMembersRow.tsx

```diff
@@ -24,159 +19,104 @@ function TeamMembersRow(props: {
-          updateMemberRole={updateMemberRole}
+          disabled={isSelf || !hasWriteAccess}
           organization={organization}
           team={team}
           member={member}
+          onChangeTeamRole={newRole => updateMemberRole(member, newRole)}
         />
-      </div>
+      </RoleSelectWrapper>
       <div>
         <RemoveButton
           hasWriteAccess={hasWriteAccess}
           hasOrgRoleFromTeam={team.orgRole !== null}
           isOrgOwner={isOrgOwner}
+          isSelf={isSelf}
           onClick={() => removeMember(member)}
           member={member}
-          user={user}
         />
       </div>
     </TeamRolesPanelItem>
   );
 }
 
-function TeamRoleSelect(props: {
-  hasWriteAccess: boolean;
-  member: TeamMember;
-  organization: Organization;
-  team: Team;
-  updateMemberRole: (member: TeamMember, newRole: string) => void;
-}) {
-  const {hasWriteAccess, organization, team, member, updateMemberRole} = props;
-  const {orgRoleList, teamRoleList, features} = organization;
-  if (!features.includes('team-roles')) {
-    return null;
-  }
-
-  // Determine the team-role, including if the current team has an org role
-  // and if adding the user to the current team changes their minimum team-role
-  const possibleOrgRoles = [member.orgRole];
-  if (member.orgRolesFromTeams && member.orgRolesFromTeams.length > 0) {
-    possibleOrgRoles.push(member.orgRolesFromTeams[0].role.id);
-  }
-  if (team.orgRole) {
-    possibleOrgRoles.push(team.orgRole);
-  }
-  const effectiveOrgRole = getEffectiveOrgRole(possibleOrgRoles, orgRoleList);
-
-  const teamRoleId = member.teamRole || effectiveOrgRole?.minimumTeamRole;
-  const teamRole = teamRoleList.find(r => r.id === teamRoleId) || teamRoleList[0];
-
-  if (
-    !hasWriteAccess ||
-    hasOrgRoleOverwrite({orgRole: effectiveOrgRole?.id, orgRoleList, teamRoleList})
-  ) {
-    return (
-      <RoleName>
-        {teamRole.name}
-        <IconWrapper>
-          <RoleOverwriteIcon
-            orgRole={effectiveOrgRole?.id}
-            orgRoleList={orgRoleList}
-            teamRoleList={teamRoleList}
-          />
-        </IconWrapper>
-      </RoleName>
-    );
-  }
-
-  return (
-    <RoleSelectWrapper>
-      <RoleSelectControl
-        roles={teamRoleList}
-        value={teamRole.id}
-        onChange={option => updateMemberRole(member, option.value)}
-        disableUnallowed
-      />
-    </RoleSelectWrapper>
-  );
-}
-
 function RemoveButton(props: {
   hasOrgRoleFromTeam: boolean;
   hasWriteAccess: boolean;
   isOrgOwner: boolean;
+  isSelf: boolean;
   member: TeamMember;
   onClick: () => void;
-  user: User;
 }) {
-  const {member, user, hasWriteAccess, isOrgOwner, hasOrgRoleFromTeam, onClick} = props;
+  const {member, hasWriteAccess, isOrgOwner, isSelf, hasOrgRoleFromTeam, onClick} = props;
 
-  const isSelf = member.email === user.email;
   const canRemoveMember = hasWriteAccess || isSelf;
   if (!canRemoveMember) {
-    return null;
+    return (
```

Comment:
Hiding user-interactive components is a bad pattern. 

Show the button, disable it with an explanation is a much better pattern.